docs/configuration_files.rst

```diff
@@ -0,0 +1,327 @@
+
+    # Blue population
+    skypy_galaxies = pipeline['blue_galaxies']
+
+    # Plot histograms
+    fig, axs = plt.subplots(1, 2, figsize=(9, 3))
+
+    axs[0].hist(skypy_galaxies['redshift'], bins=50, histtype='step', color='purple')
+    axs[0].set_xlabel(r'$Redshift$')
+    axs[0].set_ylabel(r'$\mathrm{N}$')
+    axs[0].set_yscale('log')
+
+    axs[1].hist(skypy_galaxies['magnitude'], bins=50, histtype='step', color='green')
+    axs[1].set_xlabel(r'$Magnitude$')
+    axs[1].set_yscale('log')
+
+    plt.show()
+
+Don’t forget to check out for more complete examples_!
+
+.. _examples: https://skypy.readthedocs.io/en/stable/examples/index.html
+
+
+YAML syntax
+-----------
+YAML_ is a file format designed to be readable by both computers and humans.
+Fundamentally, a file written in YAML consists of a set of key-value pairs.
+Each pair is written as ``key: value``, where whitespace after the ``:`` is optional.
+
+This guide introduces the main syntax of YAML relevant when writing
+a configuration file to use with ``SkyPy``. Essentially, it begins with
+definitions of individual variables at the top level, followed by the tables,
+and, within the table entries, the features of objects to simulate are included.
+Main keywords: ``parameters``, ``cosmology``, ``tables``.
+
+
+Variables
+^^^^^^^^^
+* `Define a variable`: a variable is defined as a key-value pair at the top of the file. You can define any type of variable:
+
+  .. code:: yaml
+
+      counter: 100 # An integer
+      miles: 1000.0 # A floating point
+      name: "Joy" # A string
+      mylist: [ 'abc', 789, 2.0e3 ] # A list
+
+
+* `Reference a variable`: variables can be referenced by their full name tagged with a dollar sign ``$``.
+  For example, if you previously defined a list of galaxy properties:
+
+  .. code:: yaml
+
+      galaxy_properties: [ 'OBJECT_ID', 'RA', 'DEC', 'REDSHIFT', 'FLUX', 'FLUX_ERR' ]
+
+  You could reference the variable:
+
+  .. code:: yaml
+
+      catalog: $galaxy_properties
+
+
+
+Functions
+^^^^^^^^^
+* `Call a function`: functions are defined as tuples where the first entry is the fully qualified function name tagged with and exclamation mark ``!`` and the second entry is either a list of positional arguments or a dictionary of keyword arguments.
+
+  For example, if you need to call the ``log10()`` and ``linspace()`` NumPy_ functions, then you define the following key-value pairs:
+
+  .. code:: yaml
+
+      log_of_2: !numpy.log10 [2]
+      myarray: !numpy.linspace [0, 2.5, 10]
+
+  You can also define parameters of functions with a dictionary of keyword arguments.
+  Imagine you want to compute the comoving distance for a range of redshifts and an `Astropy` Planck 2015 cosmology.
+  To run it with the `SkyPy` pipeline, call the function and define the parameters as an indented dictionary.
+
+  .. code:: yaml
+
+      comoving_distance: !astropy.cosmology.Planck15.comoving_distance
+        z: !numpy.linspace [ 0, 1.3, num=10 ]
```

Comment:
This example doesn't work as intended. You can't give both args and kwargs in our syntax. Either you give a list of args, or a dict of kwargs. Running this I get an error `TypeError: 'str' object cannot be interpreted as an integer` because it is trying to pass the string "num=10" as the third arg, but the function expects an integer. We should make this point more clearly in the text. The correct way to run this function as you intend is either:

```
      comoving_distance: !astropy.cosmology.Planck15.comoving_distance
        z: !numpy.linspace [0, 1.3, 10]
```

or

```
      comoving_distance: !astropy.cosmology.Planck15.comoving_distance
        z: !numpy.linspace {start: 0, stop: 1.3, num: 10}
```